File: LV1/1.6/erase/erase.py
n = int(input())
a = int(input(), 2)
b = int(input(), 2)
out = a + b

if n % 2 == 0 and a == b or (out+1) & out == 0 and out != 0:
    print("Deletion succeeded")
else:
    print("Deletion failed")

File: LV1/1.6/erase/erase.rb
n = gets.to_i
a = gets.to_i(2)
b = gets.to_i(2)
out = a + b

if (n % 2).zero? && a == b || ((out + 1) & out).zero? && out != 0
  puts 'Deletion succeeded'
else
  puts 'Deletion failed'
end
